crates/ruff_python_parser/src/parser/pattern.rs

```diff
@@ -625,35 +638,32 @@ impl<'src> Parser<'src> {
         let cls = match cls {
             Pattern::MatchAs(ast::PatternMatchAs {
                 name: Some(ident), ..
-            }) => {
-                if ident.is_valid() {
-                    Box::new(Expr::Name(ast::ExprName {
-                        range: ident.range(),
-                        id: ident.id,
-                        ctx: ExprContext::Load,
-                    }))
-                } else {
-                    #[allow(deprecated)]
-                    Box::new(Expr::Invalid(ast::ExprInvalid {
-                        value: self.src_text(&ident).into(),
-                        range: ident.range(),
-                    }))
+            }) => Box::new(Expr::Name(if ident.is_valid() {
+                ast::ExprName {
+                    range: ident.range(),
+                    id: ident.id,
+                    ctx: ExprContext::Load,
                 }
-            }
+            } else {
+                ast::ExprName {
+                    range: ident.range(),
+                    id: String::new(),
+                    ctx: ExprContext::Invalid,
+                }
+            })),
             Pattern::MatchValue(ast::PatternMatchValue { value, range: _ })
                 if matches!(value.as_ref(), Expr::Attribute(_)) =>
             {
                 value
             }
             pattern => {
                 self.add_error(
-                    ParseErrorType::OtherError("invalid pattern match class".to_string()),
+                    ParseErrorType::OtherError("invalid value for a class pattern".to_string()),
                     &pattern,
                 );
-                // FIXME(micha): Including the entire range is not ideal because it also includes trivia.
-                #[allow(deprecated)]
-                Box::new(Expr::Invalid(ast::ExprInvalid {
-                    value: self.src_text(pattern.range()).into(),
+                Box::new(Expr::Name(ast::ExprName {
+                    id: String::new(),
+                    ctx: ExprContext::Invalid,
                     range: pattern.range(),
```

Comment:
Same, I don't think we should convert arbitrary expressions or patterns to `ExprName` with the `ctx` `Invalid`. That can lead to issues downstream